### ISSUE TITLE
A static chaiscript::ChaiScript instance will cause the system crash

### ISSUE THREAD

**Issue by liwei-cpp** [ChaiScript/ChaiScript#555]
* Compiler Used: visual studio 2019
 * Operating System: Windows 10
 * Architecture (ARM/x86/32bit/64bit/etc): X64
 
### Expected Behavior 
Ideally, the system should works well even if I set the chaiscript::ChaiScript instance as static.

### Actual Behavior 
However, the real situation is not that. Because chaiscript::ChaiScript contains Thread_Storage, which contains thread_local instance and use it in the destructor:
```cpp
            ~Thread_Storage()
            {
              t().erase(this);
            }

            static std::unordered_map<const void*, T> &t()
            {
              thread_local std::unordered_map<const void *, T> my_t;
              return my_t;
            }
```
According to the C++ standard, the thread_local variable will be deleted before static variable. So ```my_t``` will be deleted, then ```~Thread_Storage()``` will be called. This lead to a invalid erase call, which cause the system crash when the program ends.

### Minimal Example to Reproduce Behavior
int main(int argc, char* argv[])
{
    static chaiscript::ChaiScript inst;
    return 0;
}

I don't know how to fix that issue. Any ideas?

**Comment by georgelam6** [comment]
You can't have a global chaiscript object with thread safety enabled. Disable thread safety by defining `CHAISCRIPT_NO_THREADS`.

**Comment by totalgee** [comment]
> I don't know how to fix that issue. Any ideas?

Don't use a static variable? (-;

Allocate your ChaiScript instance on the heap and hold it with a unique_ptr?

**Comment by lefticus** [comment]
Indeed, I already went through FAR TOO MUCH effort in the past to get ChaiScript working as a static object. This is not a feature that is supported.
